fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/bottomsheet/BottomSheet.kt

```diff
@@ -0,0 +1,493 @@
+                                            if (!sheetState.isVisible) {
+                                                scope.launch { sheetState.hide() }
+                                            }
+                                        }
+                                    },
+                                )
+                                .testTag(BOTTOMSHEET_HANDLE_TAG)
+                        ) {
+                            Icon(
+                                painterResource(id = com.microsoft.fluentui.drawer.R.drawable.ic_drawer_handle),
+                                contentDescription =
+                                if (sheetState.currentValue == BottomSheetValue.Expanded) {
+                                    "Minimize Button"
+                                } else {
+                                    if (sheetState.hasExpandedState && sheetState.isVisible) "Maximize Button" else null
+                                },
+                                tint = sheetHandleColor,
+                                modifier = Modifier
+                                    .clickable(enabled = sheetState.hasExpandedState) {
+                                        if (sheetState.currentValue == BottomSheetValue.Expanded) {
+                                            if (sheetState.confirmStateChange(BottomSheetValue.Shown)) {
+                                                scope.launch { sheetState.show() }
+                                            }
+                                        } else if (sheetState.hasExpandedState) {
+                                            if (sheetState.confirmStateChange(BottomSheetValue.Expanded)) {
+                                                scope.launch { sheetState.expand() }
+                                            }
+                                        }
+                                    }
+                            )
+                        }
+                    }
+                    Column(modifier = Modifier.testTag(BOTTOMSHEET_CONTENT_TAG),
+                        content = { sheetContent() })
+                }
+            }
+        }
+    }
+}
+
+private fun Modifier.bottomSheetSwipeable(
+    sheetState: BottomSheetState,
+    expandable: Boolean,
+    peekHeight: Dp,
+    fullHeight: Float,
+    sheetHeight: Float?
+): Modifier {
+    val modifier = if (sheetHeight != null) {
+        val peekHeightPx = min(dpToPx(peekHeight), fullHeight * BottomSheetOpenFraction)
+        val anchors = if (!expandable) {
+            mapOf(
+                fullHeight to BottomSheetValue.Hidden,
+                fullHeight - min(sheetHeight, peekHeightPx) to BottomSheetValue.Shown
+            )
+        } else if (sheetHeight <= peekHeightPx) {
+            mapOf(
+                fullHeight to BottomSheetValue.Hidden,
+                fullHeight - sheetHeight to BottomSheetValue.Shown
+            )
```

Comment:
Can be merged with || statement :
if (!expandable || sheetHeight <= peekHeighPx)

Reply:
current flow seem more readable as well on merging it increase no. of function execution.

Reply:
I feel reducing LOC will make it more readable overall. Would request you to reduce it. If you strongly feel against it, maybe we can consider it developer's creativity :).

Reply:
In my opinion, code clarity & efficiency of code take precedence over LOC. Hence would like to keep it like that :)